ci-operator/step-registry/sandboxed-containers-operator/gather-must-gather/sandboxed-containers-operator-gather-must-gather-commands.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "=========================================="
+echo "Sandboxed Containers Operator - Optional Must-Gather"
+echo "=========================================="
+
+# Check if must-gather is enabled
+if [[ "${ENABLE_MUST_GATHER:-true}" != "true" ]]; then
+    echo "Must-gather collection is disabled (ENABLE_MUST_GATHER=${ENABLE_MUST_GATHER:-})"
+    echo "Skipping must-gather step"
+    exit 0
+fi
+
+# Check if we should only run on failure
+if [[ "${MUST_GATHER_ON_FAILURE_ONLY:-false}" == "true" ]]; then
+    echo "Must-gather configured to run only on test failures"
+
+    # Check for test failure indicators
+    test_failed=false
+
+    # Method 1: Check for JUnit test results with failures
```

Comment:
Just a cosmetic (and little optimization) comment: you better use nested `if-then-else` statement. The way it's implemented here, suppose that first method is true, then it will still run the method 2 and 3 and it will still check `$test_failed`. With nested `if-then-else` you avoid checking all the methods and still can save one variable (`test_failed` won't be needed).